cloud/blockstore/libs/storage/partition/part_actor_flush.cpp

```diff
@@ -426,26 +442,52 @@ class TFlushBlocksVisitor final
             if (GetBlobRangeSize(Blocks, block.Meta.BlockIndex)
                     > MaxBlobRangeSize / BlockSize)
             {
-                Blobs.emplace_back(std::move(BlobContent), std::move(Blocks));
+                Blobs.emplace_back(
+                    std::move(BlobContent),
+                    std::move(Blocks),
+                    std::move(Checksums));
             }
 
             BlobContent.AddBlock({block.Content.Data(), block.Content.Size()});
-            Blocks.emplace_back(block.Meta.BlockIndex, block.Meta.CommitId, block.Meta.IsStoredInDb);
+            Blocks.emplace_back(
+                block.Meta.BlockIndex,
+                block.Meta.CommitId,
+                block.Meta.IsStoredInDb);
+
+            const ui32 checksumBoundary =
+                DiskPrefixLengthWithBlockChecksumsInBlobs / BlockSize;
+            const bool checksumsEnabled =
+                block.Meta.BlockIndex < checksumBoundary;
```

Comment:
this pattern appears all around the code maybe it's better to replace it with  common function `IsChecksumsEnabled(checksumedBlocksRange, blockIndex`)`

Reply:
the pattern is slightly different

I tried moving the common part to a function but it only increased the amount of code

better leave it as it is for now